pipeline/pipeline.py

```diff
@@ -71,11 +71,14 @@ def _run_task(self, task: AbstractTask):
         logging.info(f'Start {task.name}')
         if not self._dry_run and not task.skip:
             task.run()
-        end = timer()
-        logging.info(f'{task.name} Completed. Total time taken: {str(timedelta(seconds=end - start))}')
-        failed_task_log = logging_utils.get_error_log_file(task.action_type, task.object_type, self._working_dir)
-        if os.path.exists(failed_task_log) and os.path.getsize(failed_task_log) > 0:
-            msg = f'{task.name} has failures. Refer to {failed_task_log} to see failures. Terminating pipeline.'
-            logging.info(msg)
-            raise RuntimeError(msg)
-        self._completed_steps.write(f'{task.name}')
+            end = timer()
+            logging.info(f'{task.name} Completed. Total time taken: {str(timedelta(seconds=end - start))}')
+            failed_task_log = logging_utils.get_error_log_file(task.action_type, task.object_type, self._working_dir)
+            if os.path.exists(failed_task_log) and os.path.getsize(failed_task_log) > 0:
```

Comment:
If we want to use error logs to indicate errors (the errors that we need to fix always), how should we log warnings?

Previously, I thought we will treat all error logs as warnings. For errors, we can raise exceptions directly. My concern is treating all errors and warnings as hard errors will be too interruptive for users.

Reply:
This is a bit different from treating all errors and warnings as hard errors, because we wait for the entire task to finish. For the hard errors, it would just raise exception and stop in the middle of the task phase. For this one, we still let the task finish but before moving onto the next task, we interrupt the pipeline and ask users to look at the error log files, so this is almost the warning files. 

But I agree that this change can feel like a regression from the user experience perspective. We would like the invokers to make a discerning decisions, but because the invokers will be non-expert at times, we might want to do our best to differentiate between hard errors vs warnings and not let the pipeline fail if these are just warnings.

We might want to discuss on this together @yubing  @souravkhandelwal09

Reply:
Had a discussion with @yubing , I added all a catch all warn for errors in dbclient.